sound/soc/sof/pcm.c

```diff
@@ -351,6 +363,7 @@ static int sof_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 	case SNDRV_PCM_TRIGGER_SUSPEND:
 	case SNDRV_PCM_TRIGGER_STOP:
 		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_STOP;
+		reset_hw_params = true;
```

Comment:
please add spcm->prepared[substream->stream] = false; 
I am debugging issue: https://github.com/thesofproject/linux/issues/1057. For this case, the S3 is interrupted by RTC event, and ASoC only trigger SUSPEND to SOF in both FE & BE. There is no chance to invoke sof_suspend. Then
SOF prepare functions are invoked by ASoC. we need to set spcm->prepared[substream->stream] = false to make it work.  And you also set link_dev->link_prepared = 0 in BE trigger function, so just add it here

Reply:
@RanderWang The prepared variable will be set to 'false' in this case. We set 'reset_hw_params' here to true, and at tne end of function, we will call sof_pcm_dsp_pcm_free() and this will set prepared[substream]=false. So should be fine, right?

Reply:
yes,  I got it. I was confused by if(!ret)